src/injected/visualization/focus-indicator-creator.ts

```diff
@@ -21,110 +22,125 @@ export class FocusIndicatorCreator {
     ) {}
 
     public createFocusIndicator = (
-        items: TabbedItem[],
-        curElementIndex: number,
+        item: TabbedItem,
+        prevItem: TabbedItem,
         isLastItem: boolean,
         formatter: Formatter,
     ): FocusIndicator => {
-        const item = items[curElementIndex];
-        const centerPosition = this.centerPositionCalculator.getElementCenterPosition(item.element);
+        const centerPosition: Point = this.centerPositionCalculator.getElementCenterPosition(
+            item.element,
+        );
 
         if (centerPosition == null) {
             return;
         }
 
-        const drawerConfig: SVGDrawerConfiguration = formatter.getDrawerConfiguration(
+        const drawerConfig = formatter.getDrawerConfiguration(
             item.element,
             null,
         ) as SVGDrawerConfiguration;
 
-        const {
-            tabIndexLabel: { showTabIndexedLabel },
-            line: { showSolidFocusLine },
-        } = drawerConfig;
-
-        const circleConfiguration = isLastItem ? drawerConfig.focusedCircle : drawerConfig.circle;
-
-        const newCircle = this.svgShapeFactory.createCircle(centerPosition, circleConfiguration);
-        const newLabel =
-            isLastItem || !showTabIndexedLabel
-                ? null
-                : this.svgShapeFactory.createTabIndexLabel(
-                      centerPosition,
-                      drawerConfig.tabIndexLabel,
-                      item.tabOrder.toString(),
-                  );
+        if (isLastItem) {
+            return this.createIndicatorWithFocusedConfigurations(
```

Comment:
nit: just for readability I'd prefer to add an else here and move the rest of the logic to another method as well (maybe `createIndicatorWithNormalConfig`? Mostly just a preference thing though.

Reply:
Good point! I was thinking that too. I'll add it!